### PR TITLE
[FEATURE] Add ComposerReader to read information from composer.json

### PR DESCRIPTION
**Pull request by o-ba**

### CODE REVIEW

File: src/Filesystem/ComposerReader.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        if (!file_exists($filename)) {
+            return;
+        }
+        $content = @file_get_contents($filename);
+        if ($content === false) {
+            return;
+        }
+        $this->composerSchema = json_decode($content, true);
+        if (!$this->composerSchema || $this->composerSchema === []) {
+            throw new \InvalidArgumentException('The composer.json found is invalid!', 1610442954);
```
Comment (lolli42) [inline src/Filesystem/ComposerReader.php:35]: this should crash the application? wouldn't it make sense to throw a specific exception here, catch it somewhere to ignore that file? or is a working composer.json mandatory for us?

File: tests/Unit/Filesystem/ComposerReaderTest.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        rmdir('tmp');
+        parent::tearDown();
+    }
+
+    /**
+     * @test
+     */
+    public function throwsExceptionOnEmptyComposerJsonFile(): void
+    {
+        $this->expectExceptionCode(1610442954);
```
Comment (lolli42) [inline tests/Unit/Filesystem/ComposerReaderTest.php:40]: whenever possible, please have both ->expectExceptionCode() and ->expectException()

File: tests/Unit/Filesystem/ComposerReaderTest.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        file_put_contents(self::COMPOSER_FILE, '');
+        new ComposerReader('tmp');
+    }
+
+    /**
+     * @test
+     */
+    public function throwsExceptionOnInvalidComposerJsonFile(): void
+    {
+        $this->expectExceptionCode(1610442954);
```
Comment (lolli42) [inline tests/Unit/Filesystem/ComposerReaderTest.php:50]: see above.

File: src/Filesystem/ComposerReader.php (hunk 1 above)
Comment (o-ba) [reply src/Filesystem/ComposerReader.php]: If a project features a `composer.json`, it should be a valid one. We will add more automatic detection in the future. Therefore, we have to ensure we deal with a valid file here. I however added a dedicated exception which we can then catch if needed.

File: tests/Unit/Filesystem/ComposerReaderTest.php (hunk 2 above)
Comment (o-ba) [reply tests/Unit/Filesystem/ComposerReaderTest.php:41]: Done

File: tests/Unit/Filesystem/ComposerReaderTest.php (hunk 3 above)
Comment (o-ba) [reply tests/Unit/Filesystem/ComposerReaderTest.php:52]: Done

**Pull request merged**